app/controllers/vacancies_controller.rb

```diff
@@ -49,7 +49,7 @@ def set_headers
 
   def trigger_search_performed_event
     fail_safe do
-      vacancy_ids = @vacancies_search.vacancies.map(&:id)
+      vacancy_ids = @vacancies.pluck(:id)
```

Comment:
This change:
- following pagination, sends the current page vacancies instead of all the vacancies. 
- avoids loading the vacancies in memory to loop over to extract the ID. I've locally tested that the `pluck` call works without issues when called over the paginated vacancies.